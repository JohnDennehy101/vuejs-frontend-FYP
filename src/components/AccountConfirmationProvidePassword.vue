<template>
  <div class="form-control">
    <label for="password">Password</label>
    <input
      id="password"
      name="password"
      type="password"
      placeholder="Enter password"
      v-model="password"
    />
  </div>
  <div class="form-control">
    <button @click="updateUser">Submit</button>
  </div>
</template>

<script>
export default {
  data() {
    return {
      password: "",
    };
  },
  methods: {
    updateUser(event) {
      event.preventDefault();
      this.$emit("updateUserPassword", this.password);
    },
  },
};
</script>

<style scoped lang="scss">
.form-control {
  width: 70%;
  margin: 0.5rem auto;

  label {
    display: block;
    margin-bottom: 8px;
    font-weight: bold;
  }
  input {
    width: 95%;
    height: 2.5rem;
    border-radius: 12px;
    border: 1px solid grey;
    padding-left: 10px;
  }
}
button {
  border-radius: 10px;
  border: none;
  width: 95%;
  text-align: center;
  height: 2.5rem;
  margin: 1rem auto;
  background-color: $primary-button-background-colour;
  color: $primary-button-text-colour;
}
button:hover {
  cursor: pointer;
}
</style>
