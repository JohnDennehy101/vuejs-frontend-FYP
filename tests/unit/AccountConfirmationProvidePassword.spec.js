import { mount, shallowMount } from "@vue/test-utils";
import AccountConfirmationProvidePassword from "@/components/AccountConfirmationProvidePassword.vue";

describe("AccountConfirmationProvidePassword.vue", () => {
  afterEach(() => {
    jest.resetAllMocks();
  });
  it("should render correctly", () => {
    const wrapper = shallowMount(AccountConfirmationProvidePassword);
    expect(wrapper).toMatchSnapshot();
  });

  it("checks that event is emitted on click of submit button", async () => {
    const wrapper = mount(AccountConfirmationProvidePassword);

    const buttonElement = wrapper.find('button');
    await buttonElement.trigger("click");
    expect(wrapper.emitted()).toHaveProperty(
      "updateUserPassword"
    );
  });

  it("checks that input field updates on user input", async () => {
    const wrapper = mount(AccountConfirmationProvidePassword);

    const inputElement = wrapper.find("#password");
    await inputElement.setValue("ReallyHardPassword")
    expect(wrapper.vm.password).toBe("ReallyHardPassword");
  });
});
