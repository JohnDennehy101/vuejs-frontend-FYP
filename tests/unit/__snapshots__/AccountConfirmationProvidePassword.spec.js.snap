// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountConfirmationProvidePassword.vue should render correctly 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      
      <div
        class="form-control"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          placeholder="Enter password"
          type="password"
        />
        <!--v-if-->
      </div>
      <div
        class="form-control"
      >
        <button>
          Submit
        </button>
      </div>
      
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {
        "VTU_COMPONENT": Object {
          "password": "",
          "updateUser": [Function],
          "validPassword": false,
        },
      },
      "da": null,
      "data": Object {},
      "devtoolsRawSetupState": Object {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": Object {
            "password": "",
            "updateUser": [Function],
            "validPassword": false,
          },
          "_value": Object {
            "password": "",
            "updateUser": [Function],
            "validPassword": false,
          },
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {
        "VTU_COMPONENT": Object {
          "password": "",
          "updateUser": [Function],
          "validPassword": false,
        },
      },
      "refs": Object {
        "VTU_COMPONENT": Object {
          "password": "",
          "updateUser": [Function],
          "validPassword": false,
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": Array [],
        "detached": true,
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": Object {
        "VTU_COMPONENT": Object {
          "password": "",
          "updateUser": [Function],
          "validPassword": false,
        },
      },
      "slots": Object {},
      "slotsProxy": null,
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "__v_isReadonly": 0,
            "__v_isRef": 0,
            "__v_isShallow": 0,
            "__v_raw": 0,
            "__v_skip": 4,
            "_isMockFunction": 0,
            "constructor": 0,
            "isVueInstance": 0,
            "password": 2,
            "toJSON": 0,
            "updateUser": 4,
            "validPassword": 4,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "attrsProxy": null,
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "password": "",
            "updateUser": [Function],
            "validPassword": false,
          },
          "da": null,
          "data": Object {
            "password": "",
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
                ReactiveEffect {
                  "active": true,
                  "computed": ComputedRefImpl {
                    "__v_isReadonly": false,
                    "__v_isRef": true,
                    "_cacheable": true,
                    "_dirty": false,
                    "_setter": [Function],
                    "_value": false,
                    "dep": Set {
                      [Circular],
                    },
                    "effect": [Circular],
                  },
                  "deps": Array [
                    [Circular],
                  ],
                  "fn": [Function],
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {},
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {},
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "password": "",
            "updateUser": [Function],
            "validPassword": false,
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [
            [Function],
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": Array [],
            "detached": true,
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": false,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": false,
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": Array [
                        Set {
                          [Circular],
                          [Circular],
                        },
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": Array [
                  Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": Array [
                        [Circular],
                        Set {
                          [Circular],
                        },
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                    ReactiveEffect {
                      "active": true,
                      "computed": ComputedRefImpl {
                        "__v_isReadonly": false,
                        "__v_isRef": true,
                        "_cacheable": true,
                        "_dirty": false,
                        "_setter": [Function],
                        "_value": false,
                        "dep": Set {
                          [Circular],
                        },
                        "effect": [Circular],
                      },
                      "deps": Array [
                        [Circular],
                      ],
                      "fn": [Function],
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "slotsProxy": null,
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": ,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "ce": undefined,
                    "children": "Password",
                    "component": null,
                    "ctx": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <label
                      for="password"
                    >
                      Password
                    </label>,
                    "key": null,
                    "patchFlag": -1,
                    "props": Object {
                      "for": "password",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "label",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "ctx": [Circular],
                    "dirs": Array [
                      Object {
                        "arg": undefined,
                        "dir": Object {
                          "beforeUpdate": [Function],
                          "created": [Function],
                          "getSSRProps": [Function],
                          "mounted": [Function],
                        },
                        "instance": Object {
                          "password": "",
                          "updateUser": [Function],
                          "validPassword": false,
                        },
                        "modifiers": Object {},
                        "oldValue": undefined,
                        "value": "",
                      },
                    ],
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <input
                      id="password"
                      name="password"
                      placeholder="Enter password"
                      type="password"
                    />,
                    "key": null,
                    "patchFlag": 512,
                    "props": Object {
                      "id": "password",
                      "name": "password",
                      "onUpdate:modelValue": [Function],
                      "placeholder": "Enter password",
                      "type": "password",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "input",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "v-if",
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": Array [],
                    "dynamicProps": null,
                    "el": <!--v-if-->,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(v-cmt),
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="form-control"
                >
                  <label
                    for="password"
                  >
                    Password
                  </label>
                  <input
                    id="password"
                    name="password"
                    placeholder="Enter password"
                    type="password"
                  />
                  <!--v-if-->
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": Object {
                  "class": "form-control",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Submit",
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <button>
                      Submit
                    </button>,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {
                      "onClick": [Function],
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "button",
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="form-control"
                >
                  <button>
                    Submit
                  </button>
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": Object {
                  "class": "form-control",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": Array [
                  Object {
                    "arg": undefined,
                    "dir": Object {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "getSSRProps": [Function],
                      "mounted": [Function],
                    },
                    "instance": Object {
                      "password": "",
                      "updateUser": [Function],
                      "validPassword": false,
                    },
                    "modifiers": Object {},
                    "oldValue": undefined,
                    "value": "",
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <input
                  id="password"
                  name="password"
                  placeholder="Enter password"
                  type="password"
                />,
                "key": null,
                "patchFlag": 512,
                "props": Object {
                  "id": "password",
                  "name": "password",
                  "onUpdate:modelValue": [Function],
                  "placeholder": "Enter password",
                  "type": "password",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "v-if",
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": Array [],
                "dynamicProps": null,
                "el": <!--v-if-->,
                "key": null,
                "patchFlag": 0,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 8,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(v-cmt),
              },
            ],
            "dynamicProps": null,
            "el": ,
            "key": null,
            "patchFlag": 64,
            "props": null,
            "ref": null,
            "scopeId": null,
            "shapeFlag": 16,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": Symbol(v-fgt),
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "computed": Object {
              "validPassword": [Function],
            },
            "data": [Function],
            "methods": Object {
              "updateUser": [Function],
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "computed": Object {
            "validPassword": [Function],
          },
          "data": [Function],
          "methods": Object {
            "updateUser": [Function],
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.3.4",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": Array [],
  "componentVM": Object {
    "password": "",
    "updateUser": [Function],
    "validPassword": false,
  },
  "isDisabled": [Function],
  "rootVM": Object {
    "VTU_COMPONENT": Object {
      "password": "",
      "updateUser": [Function],
      "validPassword": false,
    },
  },
  "wrapperElement": ,
}
`;
